src/cmd/controller/main.go

```diff
@@ -55,63 +89,190 @@ func init() {
+	if err != nil {
+		logger.With("error", err).Fatal("failed to create terminator SQS message client builder")
+	}
+
+	cordonDrainerBuilder, err := kubectlcordondrainer.NewBuilder(
+		clientSet,
+		kubectlcordondrainer.DefaultCordoner,
+		kubectlcordondrainer.DefaultDrainer,
+	)
+	if err != nil {
+		logger.With("error", err).Fatal("failed to create kubectl cordon/drain client")
+	}
+	terminatorCordonDrainerBuilder, err := terminator.NewCordonDrainerBuilder(cordonDrainerBuilder)
+	if err != nil {
+		logger.With("error", err).Fatal("failed to create terminator cordon/drain client")
+	}
+
+	rec := terminator.Reconciler{
+		Name:                 "terminator",
+		RequeueInterval:      time.Duration(10) * time.Second,
+		NodeGetter:           nodeGetter,
+		NodeNameGetter:       nodeNameGetter,
+		SqsClientBuilder:     terminatorSqsClientBuilder,
+		SqsMessageParser:     sqsMessageParser,
+		Getter:               terminatorGetter,
+		CordonDrainerBuilder: terminatorCordonDrainerBuilder,
+	}
+	if err = rec.BuildController(
+		ctrl.NewControllerManagedBy(mgr).
+			WithOptions(controller.Options{MaxConcurrentReconciles: 10}),
+	); err != nil {
+		logger.With("error", err).
+			With("name", rec.Name).
+			Fatal("failed to create controller")
 	}
 	//+kubebuilder:scaffold:builder
 
 	if err := mgr.AddHealthzCheck("healthz", healthz.Ping); err != nil {
-		ctrlLogger.With("error", err).Fatal("unable to set up health check")
+		logger.With("error", err).Fatal("failed to set up health check")
 	}
 	if err := mgr.AddReadyzCheck("readyz", healthz.Ping); err != nil {
-		ctrlLogger.With("error", err).Fatal("unable to set up ready check")
+		logger.With("error", err).Fatal("failed to set up ready check")
 	}
 
-	ctrlLogger.Info("starting manager")
-	if err := mgr.Start(ctrl.SetupSignalHandler()); err != nil {
-		ctrlLogger.With("error", err).Fatal("problem running manager")
+	if err := mgr.Start(ctx); err != nil {
+		logger.With("error", err).Fatal("failure from manager")
 	}
 }
+
+func parseOptions() Options {
+	options := Options{}
+
+	flag.StringVar(&options.MetricsAddr, "metrics-bind-address", ":8080", "The address the metric endpoint binds to.")
+	flag.StringVar(&options.ProbeAddr, "health-probe-bind-address", ":8081", "The address the probe endpoint binds to.")
+	flag.BoolVar(&options.EnableLeaderElection, "leader-elect", false, "Enable leader election for controller manager. "+
+		"Enabling this will ensure there is only one active controller manager.")
+	flag.Parse()
+
+	return options
+}
+
+func newAwsSession() (*session.Session, error) {
+	config := aws.NewConfig().WithSTSRegionalEndpoint(endpoints.RegionalSTSEndpoint)
+	sess, err := session.NewSessionWithOptions(session.Options{
+		Config:            *config,
+		SharedConfigState: session.SharedConfigEnable,
+	})
+	if err != nil {
+		return nil, fmt.Errorf("failed to create AWS session: %w", err)
+	}
+
+	region, err := ec2metadata.New(sess).Region()
```

Comment:
Seems like this is requiring IMDS to be available which won't always be the case. It should only try IMDS as a last resort after the SDK has tried to find the region in all the well known places.